orm_lib/inc/drogon/orm/DbClient.h

```diff
@@ -29,6 +29,84 @@
 
 namespace drogon
 {
+template <typename>
+struct func_param;
+template <typename Ret, typename... Args>
+struct func_param<Ret(Args...)>
+{
+    using type = std::tuple<Args...>;
+};
+
+template <typename FuncSuccess, typename FuncFail>
+struct JSStyleFuture
+    : public std::enable_shared_from_this<JSStyleFuture<FuncSuccess, FuncFail>>
+{
+    // Hold a shared_ptr of ourselves to prevent us going out of scope
+    // before the callback is called
+    JSStypeFuture() : self_(share_from_this())
```

Comment:
You can't call share_from_this() in constructor becuase the `this` is not managed by shared_ptr here.